docs/utilities.adoc

```diff
@@ -37,3 +37,82 @@ include::{sourcedir}/utilities/MutableClockDocSpec.groovy[tag=age-filter-spec]
+verify the individual results. If any of the results is not available when `await` is called, it will also fail.
+The maximum amount of seconds to wait for can be specified as a parameter, see the example below. The default is `1.0`
+seconds. Additionally, the number of expected evaluations must be provided initially, the default value is `1`.
+
+=== Example
+
+.Test
+[source,groovy,indent=0]
+----
+include::{sourcedir}/utilities/concurrent/AsyncConditionsDocSpec.groovy[tag=async-conditions-spec]
+----
+<1> create a default `AsyncConditions` object (expecting a single evaluation)
+<2> A new thread is created, and code is passed to be run from it. This could also happen implicitly, when working with
+a method expecting a callback parameter.
+<3> pass any code that wants to do assertions to the `evaluate` method
+<4> finally, call the `await` method
+
+<5> create an `AsyncConditions` object (expecting three evaluations)
+<6> call `evaluate` multiple times
+<7> call `await` in the end, specifying 2.5 seconds as the timeout
+
+For more information have a look at the test code `spock.util.concurrent.AsyncConditionsSpec`.
+
+[[polling-conditions]]
+== Polling until conditions are fulfilled with `PollingConditions`
+
+If you need more fine grained control over asynchronous evaluation, the `PollingConditions` utility can be used in a
+similar way to `AsyncConditions`. `PollingConditions` allows to configure the timeout, delay and a delay increasing factor
+for the check. A single block of code is passed to the `eventually` method, which will evaluate this code, according to
+given time parameters, and either pass or throw one of the appropriate Spock exceptions.
+The `within` method can be used to override the timeout for a single invocation.
+There are some caveats when using this class with the `def` keyword, see the test code `spock.util.concurrent.PollingConditionsSpec`.
+
+=== Example
+
+.Test
+[source,groovy,indent=0]
+----
+include::{sourcedir}/utilities/concurrent/PollingConditionsDocSpec.groovy[tag=polling-conditions-spec]
+----
+<1> create a `PollingConditions` object, avoid the `def` keyword (see comment above)
+<2> values are set to variables in an asynchronous way
+<3> block of code containing assertions, passed in a `when` step
+<4> verify the result by checking for exceptions
+<5> only the last failed evaluation will be reported
+
+[[blocking-variables]]
+== Evaluating asynchronous variables with `BlockingVariable` and `BlockingVariables`
+
+The two utility classes `BlockingVariable` and `BlockingVariables` are there to help with collecting variables that are
+written asynchronously, e.g. from another thread. Reading the variable(s) will block the current thread until a value is
+available and returned.
+For a single variable, a type parameter can be provided, see the example below.
+The maximum amount of seconds to wait for can be specified as a parameter, the default is `1.0` seconds.
+
```

Comment:
`BlockingVariables` is not really explained here, e.g. that you have dynamic properties and how it works.